File: backend/src/main/scala/microservices/operations/hotel/objects/ManagerHotelPlannerResponse.scala
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ManagerHotelPlannerResponse(
    hotelId: String,
    hotelName: String,
    location: String,
    status: String,
    createdAt: String,
    roomTypes: List[ManagerHotelRoomTypePlannerResponse]
)
object ManagerHotelPlannerResponse:
  given sourceEncoder: Encoder[ManagerHotelPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[ManagerHotelPlannerResponse] = deriveDecoder

File: frontend/src/microservices/operations/hotel/objects/ManagerHotelPlannerResponse.ts
import type { ManagerHotelRoomTypePlannerResponse } from './ManagerHotelRoomTypePlannerResponse'

export type ManagerHotelPlannerResponse = {
  hotelId: string
  hotelName: string
  location: string
  status: string
  createdAt: string
  roomTypes: ManagerHotelRoomTypePlannerResponse[]
}

export const managerHotelPlannerResponseFromJson = (json: string): ManagerHotelPlannerResponse =>
  JSON.parse(json) as ManagerHotelPlannerResponse

export const managerHotelPlannerResponseToJson = (value: ManagerHotelPlannerResponse): string =>
  JSON.stringify(value)
